util/src/journaldb/archivedb.rs

```diff
@@ -130,7 +130,7 @@ impl HashDB for ArchiveDB {
 impl JournalDB for ArchiveDB {
 	fn spawn(&self) -> Box<JournalDB> {
 		Box::new(ArchiveDB {
-			overlay: MemoryDB::new(),
+			overlay: self.overlay.clone(),
```

Comment:
in principle because clone is meant to clone the struct, including any state that has not yet been committed to the backing DB.

in practice because when we want to clone this struct, it's because we want a backup prior to a DB commit that, if necessary we can alter and/or commit to DB later.